src/main/java/com/googlecode/lanterna/gui2/MultiWindowTextGUI.java

```diff
@@ -46,12 +46,17 @@
     private final WindowManager windowManager;
     private final BasePane backgroundPane;
     private final List<Window> windows;
+    private final List<Window> stableOrderingOfWindows;
```

Comment:
The purpose of the list stableOrderingOfWindows vs the windows list is that the WindowManager would prefer to have the same ordering everytime.

For example a 2x2 grid style window manager would like to arrange the windows in position and size and have those remain regardless of which window is active.
Prior to this change, the windows are coming through in different order because the active window is put to the front of the list, this would force most window managers to do this bookkeeping work which seems to involve overriding multiple methods.

So essentially z-order is what the existing windows list does, but most window managers would probably want access to z-ordering only secondarily to their primary ordering for layout purposes.

Reply:
Ok, but I think in that case rather than having two lists, we should create a `WindowList` class or something like that, which can encapsulate the z-index and the stable index. It would make the code interacting with the list(s) of windows in here much easier.

Reply:
I've added a WindowList class.